x-pack/plugins/alerting/server/routes/create_rule.ts

```diff
@@ -115,6 +111,21 @@ export const createRuleRoute = (
           const rulesClient = context.alerting.getRulesClient();
           const rule = req.body;
           const params = req.params;
+
+          if (params?.id) {
```

Comment:
Minor, but is there a way to reuse this code in both new and legacy routes, especially considering we're using strings a few times?

Reply:
Good call! Updated in https://github.com/elastic/kibana/pull/108572/commits/38c2c4b2e0712c0b2fe29f6880985a604b061d64